app/controllers/ResourceIndex.java

```diff
@@ -323,6 +333,9 @@ public Result readDefault(String id, String version) throws IOException {
 
 
   public Result read(String id, String version, String extension) throws IOException {
+
```

Comment:
I'm fine with this for now. We should split this 200-lines method into smaller units within the next commit.